apps/console/src/features/sms-providers/pages/sms-providers.tsx

```diff
@@ -0,0 +1,712 @@
+            }
+            if (!(values?.contentType == ContentType.JSON || values?.contentType == ContentType.FORM)) {
+                error.contentType = t(
+                    "extensions:develop.smsProviders.form.custom.validations.contentTypeInvalid"
+                );
+            }
+            if (!values?.payload) {
+                error.payload = t(
+                    "extensions:develop.smsProviders.form.custom.validations.required"
+                );
+            }
+            if (!values?.provider) {
+                error.key = t(
+                    "extensions:develop.smsProviders.form.custom.validations.required"
+                );
+            }
+        }
+
+        return error;
+    };
+
+    const handleConfigurationDelete = async (deleteBeforeUpdate?: boolean): Promise<boolean> => {
+        !deleteBeforeUpdate && setIsDeleting(true);
+
+        return deleteSMSProviders()
+            .then(() => {
+                !deleteBeforeUpdate && handleDeleteSuccess();
+                !deleteBeforeUpdate && setSmsProviderSettings({ providerParams: {}, 
+                    selectedProvider: SMSProviderConstants.TWILIO_SMS_PROVIDER });
+                !deleteBeforeUpdate && mutateSMSProviderConfig();
+
+                return true;
+            })
+            .catch((e: IdentityAppsApiException) => {
+                !deleteBeforeUpdate && handleDeleteError();
+                if (deleteBeforeUpdate && e.response.status === 404) {
+                    return true;
+                }
+
+                return false;
+            }).finally(() => {
+                setIsDeleting(false);
+            });
+    };
+
+
+    const handleDeleteError = () => {
+        dispatch(
+            addAlert({
+                description: t("extensions:develop.smsProviders." +
+                    "notifications.deleteConfiguration.error.description"),
+                level: AlertLevels.ERROR,
+                message: t("extensions:develop.smsProviders." +
+                    "notifications.deleteConfiguration.error.message")
+            })
+        );
+    };
+
+    /**
+    * Renders the loading placeholder.
+    */
+    const renderLoadingPlaceholder = () => {
+        return (
+            <div data-componentid={ `${componentId}-form-loading` }>
+                {
+                    [ ...Array(3) ].map((key: number) => {
+                        return (
+                            <Placeholder key={ key }>
+                                <Placeholder.Line length="very short" />
+                                <div>
+                                    <Placeholder.Line length="long" />
+                                    <Placeholder.Line length="medium" />
+                                </div>
+                            </Placeholder>
+                        );
+                    })
+                }
+            </div>
+        );
+    };
+
+    const providerCards: SMSProviderCardInterface[] = [
```

Comment:
Let's move this to a separate file like a config instead of putting it inside the component